src/Core/Jobs/BaseJobsHostedService.cs

```diff
@@ -142,7 +145,10 @@ public virtual async Task StartAsync(CancellationToken cancellationToken)
 
     public virtual async Task StopAsync(CancellationToken cancellationToken)
     {
-        await _scheduler?.Shutdown(cancellationToken);
```

Comment:
If `_scheduler` is null, this just throws a nullref exception as you can't await null.